doc/tutorial/sequence_file/index.md

```diff
@@ -229,98 +233,96 @@ QUAL: IIIIHHGIIIIHHGIIIH
-For this purpose, you can specialise the seqan3::sequence_file_input::selected_fields type by giving an
-additional seqan3::fields object to the constructor.
+In the examples above, we always use `auto` to deduce the record type automatically.
+In case you need the type explicitly , e.g. if you want to store the records in a variable,
+you can access the type member seqan3::sequence_file_input::record_type.
 
-\attention The **order** of the field tags in your seqan3::fields object will determine
-           the order of values stored in the record type!
+\snippet doc/tutorial/sequence_file/sequence_file_snippets.cpp record_type
 
-\snippet doc/tutorial/sequence_file/sequence_file_snippets.cpp custom_fields
+You can move the data out of the file if you want to store it somewhere without copying.
 
-<a name="assignment_reading_seq_qual"></a>
-\assignment{Exercise: Reading a FASTQ file into a SEQ_QUAL object}
+\snippet doc/tutorial/sequence_file/sequence_file_snippets.cpp record_type2
 
-While working on the same FASTQ file from assignment 1,
-create a new program that reads in the `ID` and the `SEQ_QUAL` object and
-prints the sequence and quality information to the command line.
-Take a look at the seqan3::qualified type on how to handle the combination of sequence and quality information.
+\assignment{Exercise: Storing records in a std::vector}
 
-\hint
-Pipe the seqan3::view::get on the `SEQ_QUAL` object to access the sequence or quality information .
-\endhint
+Create a small program that reads in a FASTA file and stores all the records in a std::vector.
 
-\endassignment
-\solution
+After reading, print out the vector (this works directly with the seqan3::debug_stream).
 
-\snippet doc/tutorial/sequence_file/sequence_file_solution2.cpp solution
+Test your program with the following file:
 
-The code will print the following:
-```bash
-ID:  seq1
-SEQ: AGCTAGCAGCGATCG
-QUAL: IIIIIHIIIIIIIII
-ID:  seq2
-SEQ: CGATCGATC
-QUAL: IIIIIIIII
-ID:  seq3
-SEQ: AGCGATCGAGGAATATAT
-QUAL: IIIIHHGIIIIHHGIIIH
+\snippet doc/tutorial/sequence_file/sequence_file_solution2.cpp fasta_file
+
+It should print the following:
+
+```console
+[(AGCT,seq1,),(CGATCGA,seq2,)]
 ```
+Note that the quality is empty in the third tuple entry because we are reading a FASTA file.
+
+\endassignment
+\solution
+\snippet doc/tutorial/sequence_file/sequence_file_solution2.cpp solution
 \endsolution
 
 # Sequence files as views
 
 Since SeqAn files are ranges, you can also create views over files.
 This enables us to create solutions for lot of use cases using only a few lines of code.
 
-## Reading a file in batches
+## Reading a file in chunks
 
-A common use case is to read chunks from a file instead of the whole file at once.
+A common use case is to read chunks from a file instead of the whole file at once or line by line.
 
-You can do so easily on a file range by using the seqan3::view::take.
+You can do so easily on a file range by using the ranges::view::chunk.
```

Comment:
The snippet does not show the include?

Also, is chunk gonna be in the standard? Then you should add an alias and use std::view::chunk

Reply:
it did not make it into the standard. I will add this as a todo in ranges.